compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2005,7 +2006,77 @@ private void visitFunctionPointerInvocation(BLangInvocation iExpr) {
     }
 
     private void visitBuiltInMethodInvocation(BLangInvocation iExpr) {
-        result = new BLangBuiltInMethodInvocation(iExpr, iExpr.builtInMethod);
+        String funcName = iExpr.builtInMethod.getName();
+        if (BLangBuiltInMethod.ISNAN.toString().equals(funcName)) {
```

Comment:
Can't we use a switch statement with `iExpr.builtInMethod` instead of checking function name equality?

Reply:
Used a switch statement with d548812